samples/matter/common/src/diagnostic/diagnostic_logs_retention.cpp

```diff
@@ -61,6 +61,10 @@ CHIP_ERROR DiagnosticLogsRetention::PushLog(const void *data, size_t size)
 	VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
 	VerifyOrReturnError(size < mCapacity, CHIP_ERROR_BUFFER_TOO_SMALL);
 
+	/* Make sure that no-one will start write operation from different thread in the same time. */
+	VerifyOrReturnError(!mWriteInProgress, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Can we use a zephyr lock (semaphore/mutex) here similarly to matter init purposes? I guess we can wait some time for the write/read operation (using mutex) instead of dropping messages due to multi-thread operations.

Reply:
Probably yes, I will try.